src/http/server.ts

```diff
@@ -331,10 +332,21 @@ export class Server {
         if (isNaN(e.code)) {
           e.code = 500;
         }
-        return new Response(e.stack, {
-          status: e.code,
-          headers: response.headers,
-        });
+
+        if (exception) {
+          try {
+            await exception.catch(e, originalRequest, response);
+          } catch (error) {
+            if (isNaN(error.code)) {
```

Comment:
I know this is a leftover from previous code, but I detected it now.
I would rather use a class instance assertion than a property. It avoids collision.

```ts
if (isNaN(error.code) {
  // No good. The user can throw a custom error that has code as property.
  // I think a mongoose error would enter this if, with codes > 10000 ! Definitely not status codes.
}
```

```ts
if (error instanceof HttpError) {
  // We good. The class is in fact HttpError.
}
```

Reply:
Would want a comment on that from @crookse or @ebebbington

Reply:
i should've documented this, reason it's like this is because i ran into an issue where i just threw an Error without using Drash.Errors.HttpError. it had no code associated with it, so the catch block couldn't return a response with a status code. this is basically a "if an error was thrown and was not an HTTP error with an HTTP status code, then make it a 500 by default" block

Reply:
so what should I do ?

Reply:
nothing for now. going to resolve this. this will be addressed in https://github.com/drashland/drash/issues/595